### PR TITLE
Added r-stochvol as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @conda-forge/help-r Requesting review here; only win build is failing though not sure why, the error message:

```
ERROR (r-stochvol,lib/R/library/stochvol/libs/x64/stochvol.dll): $RPATH/R.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.
```

File: recipes/r-stochvol/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    - {{ cran_mirror }}/src/contrib/Archive/stochvol/stochvol_{{ version }}.tar.gz
+  sha256: 5c149912cbb774c0baccc62ee6b78fcdae378595d143a5a17c7bb56f89cd28b3
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
```
Comment (mfansler) [inline recipes/r-stochvol/meta.yaml:21]: Either skip Windows (`skip: true  # [win]`) or whitelist DLLs (as per https://github.com/conda/conda-build/pull/4786):

```suggestion
  missing_dso_whitelist:
    - '*/R.dll'        # [win]
    - '*/Rblas.dll'    # [win]
    - '*/Rlapack.dll'  # [win]

```

File: recipes/r-stochvol/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
+    - {{ compiler('cxx') }}            # [not win]
+    - {{ compiler('m2w64_cxx') }}      # [win]
+    - {{ posix }}filesystem        # [win]
```
Comment (mfansler) [inline recipes/r-stochvol/meta.yaml:29]: The `stdlib` is different on Windows; also tidy while we're here:

```suggestion
    - {{ compiler('c') }}          # [not win]
    - {{ stdlib('c') }}            # [not win]
    - {{ compiler('m2w64_c') }}    # [win]
    - {{ stdlib('m2w64_c') }}      # [win]
    - {{ compiler('cxx') }}        # [not win]
    - {{ compiler('m2w64_cxx') }}  # [win]
    - {{ posix }}filesystem        # [win]
```

File: recipes/r-stochvol/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+test:
+  commands:
+    - $R -e "library('stochvol')"           # [not win]
+    - "\"%R%\" -e \"library('stochvol')\""  # [win]
+
+about:
+  home: https://gregorkastner.github.io/stochvol/
+  dev_url: https://github.com/gregorkastner/stochvol
+  license: GPL-2.0-only
```
Comment (mfansler) [inline recipes/r-stochvol/meta.yaml:57]: License shows `GPL (>= 2)`, so:

```suggestion
  license: GPL-2.0-or-later
```

File: recipes/r-stochvol/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  summary: "Efficient algorithms for fully Bayesian estimation of stochastic volatility (SV)
+    models with and without asymmetry (leverage) via Markov chain Monte Carlo (MCMC)
+    methods. Methodological details are given in Kastner and Fr\xFChwirth-Schnatter
+    (2014) <doi:10.1016/j.csda.2013.01.002> and Hosszejni and Kastner (2019) <doi:10.1007/978-3-030-30611-3_8>;
+    the most common use cases are described in Hosszejni and Kastner (2021) <doi:10.18637/jss.v100.i12>
+    and Kastner (2016) <doi:10.18637/jss.v069.i05> and the package examples."
+  license_family: GPL2
+  license_file:
+    - '{{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2'
```
Comment (mfansler) [inline recipes/r-stochvol/meta.yaml:66]: Include v3:

```suggestion
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-3
```

File: recipes/r-stochvol/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+# Encoding: UTF-8
```
Comment (mfansler) [inline recipes/r-stochvol/meta.yaml:67]: tidy

```suggestion
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for adding and apologies for the delay!

There are some things to touch up before merging.

**Comment by thewchan** [comment]
Thanks @mfansler @conda-forge/staged-recipes @conda-forge/help-r This is now ready for another review

**Review by mfansler — APPROVED** [review]

**Pull request merged**